src/components/modal/modal.css

```diff
@@ -0,0 +1,35 @@
+.modal-overlay {
+  background: rgba(0, 0, 0, 0.4);
+  backdrop-filter: blur(2px);
+  width: 100vw;
+  height: 100vh;
+  z-index: 99999;
+  position: fixed;
```

Comment:
Suggestions: Z-index can me defined in variables, it will help in fixing problems with stacking context if arises.